### PR TITLE
Tel 104 make hero component use packet context data

### PR DESCRIPTION
**Pull request by alexwhelan12**

### CODE REVIEW

File: client/src/components/atoms/SpeedAtom.tsx [hunk 1]
```diff
@@ -1,15 +1,20 @@
+import { useEffect, useState } from "react";
+
 import { useAppState } from "@/contexts/AppStateContext";
+import { usePacket } from "@/contexts/PacketContext";
 
-function SpeedAtom(props: any) {
+function SpeedAtom() {
+  const { currentPacket } = usePacket();
   const {} = useAppState;
```
Comment (ideen1) [inline client/src/components/atoms/SpeedAtom.tsx:8]: This code is not needed, we can remove it.

File: client/src/components/atoms/ThrottleIcon.tsx [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
-  const [regenPos, setRegenPos] = useState<number>(0);
-  const [gasPos, setGasPos] = useState<number>(62);
+import { usePacket } from "@/contexts/PacketContext";
+
+function ThrottleIcon() {
+  const { currentPacket } = usePacket();
+  const gasPos = currentPacket.DriverControls.Acceleration;
+  const regenPos = currentPacket.DriverControls.RegenBraking;
 
   const [throttleStyleString, setThrottleStyleString] = useState<string>("");
```
Comment (ideen1) [inline client/src/components/atoms/ThrottleIcon.tsx:10]: This code is not needed, we can remove it.

File: client/src/components/atoms/ThrottleIcon.tsx [hunk 3]
```diff
@@ -1,32 +1,48 @@
...
+  // useEffect(() => {
+  //   const nextString = `linear-gradient(
+  //           90deg,
+  //           rgba(0,0,0,0) 0%, rgba(0,0,0,0) ${50 - regenPos / 2}%,
+  //           #96be25 ${50 - regenPos / 2}%, #96be25 50%,
+  //           #B94A6C 50%, #B94A6C ${50 + gasPos / 2}%,
+  //           rgba(0,0,0,0) ${50 + gasPos / 2}%, rgba(0,0,0,0) 100%
+  //           )`;
+  //   setThrottleStyleString(nextString);
+  // }, [regenPos, gasPos]);
```
Comment (ideen1) [inline client/src/components/atoms/ThrottleIcon.tsx:21]: This code is not needed, we can remove it.

File: client/src/components/molecules/HeroMolecules/BatteryThrottleComponent.tsx [hunk 4]
```diff
@@ -1,16 +1,15 @@
 import BatteryIconComponent from "@/components/atoms/BatteryIcon";
 import SpeedAtom from "@/components/atoms/SpeedAtom";
 import ThrottleIcon from "@/components/atoms/ThrottleIcon";
+import type ITelemetryData from "@/objects/telemetry-data.interface";
```
Comment (ideen1) [inline client/src/components/molecules/HeroMolecules/BatteryThrottleComponent.tsx:4]: This code is not needed, we can remove it.

File: client/src/components/molecules/HeroMolecules/CarGraphicComponent.tsx [hunk 5]
```diff
@@ -7,6 +7,8 @@ import { CarModelComponent } from "@/components/molecules/HeroMolecules/CarMolec
 import { RoadComponent } from "@/components/molecules/HeroMolecules/CarMolecules/RoadComponent";
 import type { IndicationLocations } from "@/components/molecules/HeroMolecules/HeroTypes";
 import { IndicationStates } from "@/components/molecules/HeroMolecules/HeroTypes";
+import { usePacket } from "@/contexts/PacketContext";
+import type ITelemetryData from "@/objects/telemetry-data.interface";
```
Comment (ideen1) [inline client/src/components/molecules/HeroMolecules/CarGraphicComponent.tsx:11]: This code is not needed, we can remove it.

File: client/next.config.mjs [hunk 6]
```diff
@@ -1,5 +1,32 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  webpack(config) {
```
Comment (ideen1) [inline client/next.config.mjs:3]: Please undo changes in this file, theya re no longer needed.

File: package.json [hunk 7]
```diff
@@ -51,6 +51,7 @@
     "@aws-amplify/backend-output-schemas": "^0.6.0",
     "@aws-amplify/data-schema-types": "^0.7.5",
     "@aws-amplify/plugin-types": "^0.8.0",
+    "@svgr/webpack": "^8.1.0",
```
Comment (ideen1) [inline package.json:54]: We can uninstall this library now that we no longer need it.

File: .vscode/settings.json [hunk 8]
```diff
@@ -12,5 +12,8 @@
     "editor.formatOnSave": true
   },
   "editor.defaultFormatter": "esbenp.prettier-vscode",
-  "editor.formatOnSave": true
+  "editor.formatOnSave": true,
+  "[typescriptreact]": {
```
Comment (ideen1) [inline .vscode/settings.json:16]: Remove this change

**Review by ideen1 — CHANGES_REQUESTED** [review]
Looks great overall, just some code cleanup changes.

**Review by ideen1 — DISMISSED** [review]

**Review by JenniferJa — APPROVED** [review]

**Review by JenniferJa — DISMISSED** [review]

**Review by JenniferJa — APPROVED** [review]

**Review by ideen1 — APPROVED** [review]

**Pull request merged**